### PR TITLE
update basic-ftp package to version 4.6.6

### PR DESCRIPTION
**Pull request by kirill-ivlev**
**Task name**: FtpUploadV2
**Description**: Old version of the basic-ftp is not correctly handles connection problems, updating basic-ftp to latest version, including typescript (which required to build fresh version) as dev dependency.
In some pipelines tests run may fail after updating mocha version in #16192, according to that, this check also was updated to fix this issue.

**Documentation changes required:** N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) None

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Comment by anatolybolshakov** [comment]
@kirill-ivlev could you please describe scenarios tested?

**Comment by anatolybolshakov** [comment]
Have we also considered [possible breaking changes ](https://github.com/patrickjuchli/basic-ftp/blob/master/CHANGELOG.md) from basic-ftp v3.4.3 to 4.6.6?

**Comment by kirill-ivlev** [comment]
@anatolybolshakov 
The following scenarios were tested:
1. Error handling in case if IIS FTP limiting connection by IP _-working as expected_
1. Files uploading with FTP protocol. _-working as expected_

I reviewed the changelog and seems that no one of the listed breaking changes should affect us

File: Tasks/FtpUploadV2/task.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "demands": [],
     "version": {
         "Major": 2,
-        "Minor": 201,
+        "Minor": 205,
```
Comment (alexander-smolyakov) [inline Tasks/FtpUploadV2/task.json:21]: Pleas bump task version

**Comment by anatolybolshakov** [comment]
@kirill-ivlev seems like ci is broken - could you please take a look?

**Review by anatolybolshakov — APPROVED** [review]
Looks good for me in general, thanks!

File: make.js [hunk 2]
```diff
@@ -405,7 +405,7 @@ target.test = function() {
 
 target.testLegacy = function() {
     ensureTool('tsc', '--version', 'Version 2.3.4');
-    ensureTool('mocha', '--version', '5.2.0');
+    ensureTool('mocha', '--version', '6.2.3');
```
Comment (KonstantinTyukalov) [inline make.js:408]: Any idea why this might have been necessary?

**Review by KonstantinTyukalov — APPROVED** [review]

**Comment by kirill-ivlev** [comment]
> @kirill-ivlev seems like ci is broken - could you please take a look?

@anatolybolshakov fixed

File: make.js (hunk 2 above)
Comment (kirill-ivlev) [reply make.js]: fix was merged in #16378

**Review by anatolybolshakov — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Review by DenisRumyantsev — APPROVED** [review]

**Pull request merged**